### PR TITLE
Fill ERROR_CODE field in RobotStatus msg 

### LINKED ISSUE #3020: Fill ERROR_CODE field in RobotStatus msg
### Description of the task
We support sending error messages, such as high cpu temperature, in the RobotStatus proto msgs we send from the robot, however, they are not filled in. We should have some thresholds for when we should fill these fields. 

Should also see if there are other error messages we can/should include.

### Acceptance criteria

- [ ]

### PR DESCRIPTION
**Pull request by Andrewyx**
### Description

Added func to Thunderloop to poll errors. Sends notif once upon detection of each type.
Added Constant for Jetson max temp (97 C)
Changed Capacitor warning to trigger upon overvolt: renamed corresponding functions to keep consistency
Removed Dribbler Temps --> Redundant as in other classes: made reserved in protobufs (enum 2)

### Testing Done

Completed in-person field testing with Swolebot. Successfully displays error codes upon threshold cross.

### Resolved Issues

Resolves #3020 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

File: src/shared/constants.h [hunk 1]
```diff
@@ -171,3 +171,6 @@ static const unsigned NUM_TICKS_PER_TRAJECTORY_ELEMENT = 4u;
 static const unsigned CONTROL_LOOP_HZ                  = 60u;
 
 static const unsigned NUM_GENEVA_ANGLES = 5;
+
+// Jetson Nano Constants
+static const double MAX_JETSON_TEMP = 97;
```
Comment (itsarune) [inline src/shared/constants.h]: suffix the unit here: `MAX_JETSON_TEMP_C`

File: src/software/jetson_nano/thunderloop.h [hunk 2]
```diff
@@ -92,6 +92,11 @@ class Thunderloop
      */
     double getNanoseconds(timespec time);
 
+    /**
+     * Sends ErrorCodes for BAT, CAP, CPU TEMP, DRIBBLER TEMP if over thresholds
```
Comment (itsarune) [inline src/software/jetson_nano/thunderloop.h]: remove DRIBBLER_TEMP here from this comment

File: src/software/sensor_fusion/sensor_fusion_test.cpp [hunk 3]
```diff
@@ -269,7 +269,7 @@ class SensorFusionTest : public ::testing::Test
     {
         auto robot_msg = std::make_unique<TbotsProto::RobotStatus>();
         robot_msg->set_robot_id(2);
-        robot_msg->add_error_code(TbotsProto::ErrorCode::LOW_CAP);
+        robot_msg->add_error_code(TbotsProto::ErrorCode::HIGH_CAP);
         robot_msg->add_error_code(TbotsProto::ErrorCode::DRIBBLER_MOTOR_HOT);
```
Comment (itsarune) [inline src/software/sensor_fusion/sensor_fusion_test.cpp:273]: you can remove this line

File: src/software/sensor_fusion/sensor_fusion_test.cpp [hunk 4]
```diff
@@ -21,7 +21,7 @@ class SensorFusionTest : public ::testing::Test
           geom_data(initSSLDivBGeomData()),
           robot_status_msg_id_1(initRobotStatusId1()),
           robot_status_msg_id_2(initRobotStatusId2()),
-          robot_status_msg_low_cap(initLowCapErrorCode()),
+          robot_status_msg_high_cap(initLowCapErrorCode()),
```
Comment (itsarune) [inline src/software/sensor_fusion/sensor_fusion_test.cpp]: rename initLowCapErrorCode -> initHighCapErrorCode

File: src/software/jetson_nano/thunderloop.cpp [hunk 5]
```diff
@@ -405,3 +408,23 @@ double Thunderloop::getCpuTemperature()
         return 0.0;
     }
 }
+
+void Thunderloop::sendErrorCodes()
+{
+    // Sends error code once for threshold
+    if (power_status_.battery_voltage() <= BATTERY_WARNING_VOLTAGE && !sent_errors[0])
```
Comment (itsarune) [inline src/software/jetson_nano/thunderloop.cpp]: One issue with this implementation is that these errors won't get cleared (they get set to true but don't get set back to false).

Here's a different approach:
1. call `robot_status_.clear_error_code()`
2.  and then directly add the error code without the `sent_errors` variable

**Review by itsarune — COMMENTED** [review]
nice work! I have some suggested changes

File: src/software/sensor_fusion/sensor_fusion_test.cpp (hunk 3 above)
Comment (itsarune) [reply src/software/sensor_fusion/sensor_fusion_test.cpp:273]: (the DRIBBLER_MOTOR_HOT one)

File: src/software/jetson_nano/thunderloop.cpp [hunk 6]
```diff
@@ -405,3 +408,23 @@ double Thunderloop::getCpuTemperature()
         return 0.0;
     }
 }
+
+void Thunderloop::sendErrorCodes()
```
Comment (nimazareian) [inline src/software/jetson_nano/thunderloop.cpp]: nit: this isn't sending error codes to thunderscope, rather, it's setting/updating the error codes. I think we should rename the method. The comments here and in the `.h` file should be updated to reflect it as well.

File: src/software/jetson_nano/thunderloop.h [hunk 7]
```diff
@@ -109,6 +114,8 @@ class Thunderloop
     TbotsProto::PrimitiveExecutorStatus primitive_executor_status_;
     TbotsProto::Timestamp time_sent_;
 
+    // Error Codes
+    TbotsProto::ErrorCode error_code;
```
Comment (nimazareian) [inline src/software/jetson_nano/thunderloop.h]: nit: unused field

File: src/software/sensor_fusion/sensor_fusion_test.cpp (hunk 4 above)
Comment (nimazareian) [reply src/software/sensor_fusion/sensor_fusion_test.cpp]: CI seems to be failing due to this

**Review by nimazareian — COMMENTED** [review]
Just looking at the changes without running it, it looks good. Just had some nits.

**Comment by Andrewyx** [comment]
Oh thanks for the catches, I'll implement them right now

**Comment by Andrewyx** [comment]
Took a couple days, but found one stray test with a missing field which seemed to be causing local test errors

**Comment by Andrewyx** [comment]
New changes field tested, working well.

**Review by nimazareian — APPROVED** [review]
LGTM! 🔥

**Review by itsarune — APPROVED** [review]
LGTM!

**Review by itsarune — COMMENTED** [review]
actually, you need to pull upstream for the new CI check

File: src/software/sensor_fusion/sensor_fusion_test.cpp (hunk 3 above)
Comment (itsarune) [reply src/software/sensor_fusion/sensor_fusion_test.cpp:273]: bump: you can delete `DRIBBLER_MOTOR_HOT` from [](https://github.com/UBC-Thunderbots/Software/blob/master/src/proto/robot_status_msg.proto) line 81 and delete all the code related to `DRIBBLER_MOTOR_HOT`

**Pull request merged**